src/main/java/net/mcreator/ui/procedure/RetvalProcedureSelector.java

```diff
@@ -0,0 +1,251 @@
+		super(mcreator, returnType, providedDependencies);
+
+		this.fixedValue = fixedValue;
+		this.defaultName = L10N.t("procedure.common.fixed");
+
+		setLayout(new BorderLayout(0, 0));
+
+		addMouseListener(new MouseAdapter() {
+			@Override public void mouseClicked(MouseEvent e) {
+				e.consume();
+			}
+		});
+
+		setBorder(BorderFactory.createEmptyBorder(0, 2, 0, 2));
+
+		setOpaque(true);
+		procedures.setBorder(BorderFactory.createLineBorder(returnType.getBlocklyColor()));
+		setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
+
+		procedures.setRenderer(new ConditionalComboBoxRenderer());
+		procedures.addPopupMenuListener(new ComboBoxFullWidthPopup());
+		procedures.addActionListener(e -> {
+			CBoxEntry selectedItem = procedures.getSelectedItem();
+			if (selectedItem != null) {
+				if (!selectedItem.correctDependencies) {
+					procedures.setSelectedItem(oldItem);
+				} else {
+					oldItem = selectedItem;
+				}
+			}
+			updateDepsList(true);
+		});
+
+		JPanel top = new JPanel(new BorderLayout());
+		top.setOpaque(false);
+
+		JLabel nameLabel = new JLabel(eventName);
+		ComponentUtils.deriveFont(nameLabel, 12);
+
+		JLabel eventNameLabel = new JLabel();
+		eventNameLabel.setFont(eventNameLabel.getFont().deriveFont(9f));
+		if (side == Side.CLIENT) {
+			eventNameLabel.setIcon(UIRES.get("16px.client"));
+			eventNameLabel.setToolTipText(L10N.t("trigger.triggers_on_client_side_only"));
+			if (helpContext == null)
+				top.add("North", PanelUtils.westAndCenterElement(eventNameLabel, nameLabel));
+			else
+				top.add("North", PanelUtils.westAndCenterElement(eventNameLabel,
+						HelpUtils.wrapWithHelpButton(helpContext, nameLabel, SwingConstants.LEFT)));
+		} else if (side == Side.SERVER) {
+			eventNameLabel.setToolTipText(L10N.t("trigger.triggers_on_server_side_only"));
+			eventNameLabel.setIcon(UIRES.get("16px.server"));
+			if (helpContext == null)
+				top.add("North", PanelUtils.westAndCenterElement(eventNameLabel, nameLabel));
+			else
+				top.add("North", PanelUtils.westAndCenterElement(eventNameLabel,
+						HelpUtils.wrapWithHelpButton(helpContext, nameLabel, SwingConstants.LEFT)));
+		} else {
+			if (helpContext == null)
+				top.add("North", nameLabel);
+			else
+				top.add("North", HelpUtils.wrapWithHelpButton(helpContext, nameLabel, SwingConstants.LEFT));
+		}
+
+		top.add("South", depslab);
+
+		add.setContentAreaFilled(false);
+		add.setOpaque(false);
+		add.setMargin(new Insets(0, 0, 0, 0));
+		add.addActionListener(e -> {
+			String procedureNameString = "";
+			if (mcreator.mcreatorTabs.getCurrentTab().getContent() instanceof ModElementGUI) {
+				StringBuilder procedureName = new StringBuilder(
+						((ModElementGUI<?>) mcreator.mcreatorTabs.getCurrentTab().getContent()).getModElement()
+								.getName());
+				String[] parts = eventName.replaceAll("\\(.*\\)", "").split(" ");
+				for (String part : parts) {
+					procedureName.append(StringUtils.uppercaseFirstLetter(part));
```

Comment:
You should maybe inline this loop too.

Reply:
You can't call two methods from method referenc